src/hotspot/share/runtime/thread.hpp

```diff
@@ -643,6 +643,14 @@ class Thread: public ThreadShadow {
     assert(_wx_state == expected, "wrong state");
   }
 #endif // __APPLE__ && AARCH64
+
+ // support ASGCT
+ private:
+  bool _in_asgct;
```

Comment:
The position of this field may be significant. See if there are gaps in the Thread structure which this bool might be able to fill.

Reply:
Just to clarify - you mean finding a gap due to padding and putting the field there?
The rest of the fields are usually clustered around the supported functionality so before I insert this flag somewhere in the middle of unrelated fields I want to be sure this is what we want.

Reply:
I had tried identifying any gaps I might use but found none. Not saying they are none but it is rather difficult to spot anything with all the Thread related attributes spread across many lines, interspersed with the method declarations :(

Reply:
An option would be to place it after `jint                  _in_deopt_handler;` but it would make the code less coherent.

Reply:
At the start of JavaThread we have:
```
 private:
  bool           _on_thread_list;                // Is set when this JavaThread is added to the Threads list
  OopHandle      _threadObj;                     // The Java level thread object
```
so adding it next to the existing bool seems good. The accessors don't have to be defined at the same place.

Reply:
Ok. Done.
This requires an extra check and a static cast to `JavaThread` in `CodeCache::find_blob()` but it should not be that bad.